src/com/facebook/buck/apple/AppleLibraryDescription.java

```diff
@@ -1062,13 +1062,19 @@ private HeaderSymlinkTree createUnderlyingModuleSymlinkTreeBuildRule(
                 buildTarget, cellRoots, graphBuilder, cxxPlatform)
             ::convert);
 
-    if (SwiftDescriptions.isSwiftOnly(graphBuilder.getSourcePathResolver(), args.getSrcs())
-        && args.getExportedHeaders().isEmpty()) {
+    if (SwiftDescriptions.hasSwiftSource(graphBuilder.getSourcePathResolver(), args.getSrcs())) {
       BuildTarget swiftCompileTarget = AppleLibraryDescriptionSwiftEnhancer.createBuildTargetForSwiftCompile(buildTarget, cxxPlatform);
       SwiftCompile compile = (SwiftCompile) graphBuilder.requireRule(swiftCompileTarget);
       cxxPreprocessorInputBuilder.addIncludes(
-          CxxHeadersDir.of(CxxPreprocessables.IncludeType.LOCAL, compile.getSourcePathToOutput()));
-    } else {
+          CxxHeadersDir.of(CxxPreprocessables.IncludeType.LOCAL, compile.getSwiftModuleOutputPath()));
```

Comment:
This is the change that did the trick. Instead of adding the whole `apple-swift-compile` as a dependency, only `.swiftmodule` is added.

I think this is safe enough, I'm not sure if we should add more dependencies (like obj-c headers) if the module has mixed source code. The dev branch has some differences regarding these headers that let me think it's not necessary... we also don't have mixed source code on our repository. But it's a point of attention in we start receiving strange linker errors.

References
https://github.com/airbnb/buck/pull/23
https://github.com/facebook/buck/commit/0a6999a7514413f11c5d783df04f0bea28fed6b9